pkg/util/nativeutils/eventbus/listener.go

```diff
@@ -20,10 +22,22 @@ import (
 	"github.com/dusk-network/dusk-blockchain/pkg/util/nativeutils/hashset"
 )
 
+var (
+	// ErrMsgChanFull underlying queue fails to accept new message due to full buffer.
+	ErrMsgChanFull = errors.New("message channel buffer is full")
+
+	// ErrRingBufferClosed underlying ring buffer is closed.
+	ErrRingBufferClosed = errors.New("ringbuffer is closed")
+)
+
 // Listener publishes a byte array that subscribers of the EventBus can use.
 type Listener interface {
 	// Notify a listener of a new message.
 	Notify(message.Message) error
+
+	// Update Listeners log level. From verbose to silent.
+	SetLogLevel(logrus.Level)
```

Comment:
Replace the `SetLogLevel` method with `Setup` and `Teardown`. The `LogLevel` can be manipulated in these methods rather than being exposed. See the other comments above

Reply:
It makes sense to me to replace `SetLogLevel` with `Setup(loglevel)` method but not sure there is any need for `Listener.Teardown`. At consensus.Teardown method we need to update  ChanListener to make it temporarily silent but not to tear it down.